OriginProtocol/origin-dapp
# John/boost listing

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything

### Description:

- Adds boost slider as step 3 of create listing workflow
- Shows boost info box if it's the first time the seller has created a listing
- Otherwise, shows boost slider
- Shows warnings if user has zero OGN or if the selected boost is greater than their OGN balance
- Adds future support to `getFiatPrice()`util for OGN token price conversion

### Testing Instructions:
- in `listing-create.js` in the constructor, change `ognBalance` and `isFirstListing` to see the different states of the new Step 3
- To avoid having to fill out the form each time, you can set the step to `step: this.STEP.BOOST` in that same constructor

<img width="589" alt="screen shot 2018-08-31 at 9 39 54 am" src="https://user-images.githubusercontent.com/5402813/44922267-2e1d1d80-ad02-11e8-9203-bcf4a5892d70.png">

--------------------------------------------------------------------

<img width="276" alt="screen shot 2018-08-31 at 9 40 13 am" src="https://user-images.githubusercontent.com/5402813/44922269-2e1d1d80-ad02-11e8-9d42-1430c9e21002.png">

--------------------------------------------------------------------

<img width="594" alt="screen shot 2018-08-31 at 9 40 45 am" src="https://user-images.githubusercontent.com/5402813/44922270-2e1d1d80-ad02-11e8-9d73-b35ccc954f5e.png">

--------------------------------------------------------------------

<img width="720" alt="screen shot 2018-08-31 at 9 40 52 am" src="https://user-images.githubusercontent.com/5402813/44922271-2e1d1d80-ad02-11e8-8cb0-6f10e1c02f7d.png">

--------------------------------------------------------------------

<img width="582" alt="screen shot 2018-08-31 at 9 41 02 am" src="https://user-images.githubusercontent.com/5402813/44922272-2eb5b400-ad02-11e8-8cc2-f82bb87eb3be.png">

--------------------------------------------------------------------

<img width="596" alt="screen shot 2018-08-31 at 9 41 22 am" src="https://user-images.githubusercontent.com/5402813/44922274-2eb5b400-ad02-11e8-9dfd-4ce7e0965bb3.png">
